praw/reddit.py

```diff
@@ -381,18 +383,36 @@ def info(self, fullnames):
                   to obtain its replies, you will need to call :meth:`.refresh`
                   on the yielded :class:`.Comment`.
 
-        """
-        if not isinstance(fullnames, list):
-            raise TypeError('fullnames must be a list')
-
-        def generator():
-            for position in range(0, len(fullnames), 100):
-                fullname_chunk = fullnames[position:position + 100]
-                params = {'id': ','.join(fullname_chunk)}
-                for result in self.get(API_PATH['info'], params=params):
-                    yield result
+        .. note:: When using the url option, it is important to be aware that
+                  urls are treated literally by Reddit apis. As such, the urls
+                  "youtube.com" and "https://www.youtube.com" will provide a
+                  different set of submissions.
 
-        return generator()
+        """
+        if bool(fullnames) == bool(url):
+            raise TypeError('Either `fullnames` or `url` must be provided.')
+
+        elif bool(fullnames):
```

Comment:
Since we've already ensured that only one is set, this can simply be `elif fullnames`.